language/move-vm/runtime/src/data_cache.rs

```diff
@@ -1,75 +1,25 @@
 // Copyright (c) The Libra Core Contributors
```

Comment:
this moved from `state` to `runtime` as it was the only thing left in `state` and it was required by the `runtime`.
Users of this are users of the runtime so it does not even force people to extra dependencies.
`RemoteCache` is the only trait that could be moved elsewhere (e.g. `move-vm/types`) and we'll see.
If I could (and maybe we will be able to after @vgao1996 work on values) I would remove `DataStore` and force this concrete implementation in the API.
For now clients should use this and certainly that is what Libra uses

Reply:
My take on this: 
- Concrete implementations like TransactionDataCache belongs to the runtime so I'm definitely fine moving them there. 
- I'm not too sure about the removal of the traits and forcing TransactionDataCache onto the clients. The way I see this is that TransactionDataCache is an additional layer/adapter on top of whatever DataStore the client passes in and it's kinda internal. Leaking it to the client seems a bit weird to me at the moment. Also consider that we now have folks trying to use Move & at least part of the Move VM on other blockchains, and I think a trait will provide more flexibility AND better isolation. With that being said, I'm 100% supportive of having a concrete implementation rather than a trait for things that are in the Libra VM/Adapter.